RIOT-OS/RIOT
# congure_quic: initial import of QUIC congestion control

### Contribution description
This provides an implementation for [CongURE](https://github.com/RIOT-OS/RIOT/pull/15951) of the congestion control mechanism of the QUIC protocol as specified in [`draft-ietf-quic-recovery`](https://datatracker.ietf.org/doc/draft-ietf-quic-recovery/) (version 34 at the opening of this PR). It has the advantage of having congestion control via loss and ECN already build in and also provides means for packet pacing, so perfect for SFR!



### Testing procedure
Tests are provided for which [`riotctrl`](https://github.com/RIOT-OS/riotctrl/) (at least v0.2.2) is required for them to run. If installed,

```
make -j -C tests/congure_quic flash test
```

should succeed.



### Issues/PRs references
Requires ~~#15951~~ (merged), ~~#16111, and #16119.~~ (merged) When merged, I will adapt the test framework to use the turo framework introduced in #15950, which was the original idea to begin with (which is why it is already printing JSON strings).
![PR dependency graph](https://page.mi.fu-berlin.de/mlenders/congure.svg)
